### PR TITLE
libgpiod: added package to repo

### PR DESCRIPTION
**Pull request by Greeley**
The hash doesn't verify for any version
The sha256 of the gzip archive is(v2.0.1): 62071ac22872d9b936408e4a067d15edcdd61dce864ace8725eacdaefe23b898 (result of: `shasum -a 256 libgpiod-2.0.1.tar.gz` The sha256() function in xmake seems to come up with something different.

### CODE REVIEW

**Comment by waruqi** [comment]
please push to dev branch next.

File: packages/l/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    -- add_configs(cxx-bindings, {description = "Build c++ bindings for library.", default = "--enable-bindings-cxx", readonly = true})
+    add_cxxflags("--std=c++17")
+
+    on_install("macos", "linux", function (package)
+        local configs = {}
+        os.runv("./autogen.sh --enable-tools=yes -enable-bindings-cxx --prefix=".. package:installdir())
+    --    table.insert(configs, "-DMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+    --    table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        import("package.tools.make").make(package, configs)
+        import("package.tools.make").make(package, {"install"})
```
Comment (waruqi) [inline packages/l/xmake.lua:21]: why not use `tools.autoconf`?

File: packages/l/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    set_description("libgpiod - C library and tools for interacting with the linux GPIO character device (gpiod stands for GPIO device)")
+    
+    add_urls("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-$(version).tar.gz",
+             "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git")
+    add_versions("v2.0.1", "62071ac22872d9b936408e4a067d15edcdd61dce864ace8725eacdaefe23b898")
+    add_versions("v2.0", "1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c")
+    add_versions("v1.6.4", "9ae6a5ffa77462fee4fc34c24a7c2459a2a5851f")
+
+    add_deps("make", "autoconf", "automake", "libtool","pkg-config")
+    -- add_configs(cxx-bindings, {description = "Build c++ bindings for library.", default = "--enable-bindings-cxx", readonly = true})
```
Comment (waruqi) [inline packages/l/xmake.lua:12]: remove unused comments

File: packages/l/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    add_cxxflags("--std=c++17")
+
+    on_install("macos", "linux", function (package)
+        local configs = {}
+        os.runv("./autogen.sh --enable-tools=yes -enable-bindings-cxx --prefix=".. package:installdir())
+    --    table.insert(configs, "-DMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+    --    table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/l/xmake.lua:19]: remove comments

File: packages/l/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    add_versions("v2.0", "1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c")
+    add_versions("v1.6.4", "9ae6a5ffa77462fee4fc34c24a7c2459a2a5851f")
+
+    add_deps("make", "autoconf", "automake", "libtool","pkg-config")
+    -- add_configs(cxx-bindings, {description = "Build c++ bindings for library.", default = "--enable-bindings-cxx", readonly = true})
+    add_cxxflags("--std=c++17")
+
+    on_install("macos", "linux", function (package)
+        local configs = {}
+        os.runv("./autogen.sh --enable-tools=yes -enable-bindings-cxx --prefix=".. package:installdir())
```
Comment (waruqi) [inline packages/l/xmake.lua:17]: please use tools.autoconf

File: packages/l/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    
+    add_urls("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-$(version).tar.gz",
+             "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git")
+    add_versions("v2.0.1", "62071ac22872d9b936408e4a067d15edcdd61dce864ace8725eacdaefe23b898")
+    add_versions("v2.0", "1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c")
+    add_versions("v1.6.4", "9ae6a5ffa77462fee4fc34c24a7c2459a2a5851f")
+
+    add_deps("make", "autoconf", "automake", "libtool","pkg-config")
+    -- add_configs(cxx-bindings, {description = "Build c++ bindings for library.", default = "--enable-bindings-cxx", readonly = true})
+    add_cxxflags("--std=c++17")
```
Comment (waruqi) [inline packages/l/xmake.lua:13]: please remove it

File: packages/l/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    set_homepage("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/about/")
+    set_description("libgpiod - C library and tools for interacting with the linux GPIO character device (gpiod stands for GPIO device)")
+    
+    add_urls("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-$(version).tar.gz",
+             "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git")
+    add_versions("v2.0.1", "62071ac22872d9b936408e4a067d15edcdd61dce864ace8725eacdaefe23b898")
+    add_versions("v2.0", "1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c")
+    add_versions("v1.6.4", "9ae6a5ffa77462fee4fc34c24a7c2459a2a5851f")
+
+    add_deps("make", "autoconf", "automake", "libtool","pkg-config")
```
Comment (waruqi) [inline packages/l/xmake.lua:11]: add space ` , ` and remove make package

Comment (Greeley) [reply packages/l/xmake.lua]: removed make,
removed comments,
switched to tools.autoconf (didn't know about it)

Hope this is better!

File: packages/l/xmake.lua (hunk 1 above)
Comment (Greeley) [reply packages/l/xmake.lua]: switched to `tools.autoconf`

File: packages/l/libgpiod/xmake.lua [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    set_description("libgpiod - C library and tools for interacting with the linux GPIO character device (gpiod stands for GPIO device)")
+    
+    add_urls("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-$(version).tar.gz",
+             "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git")
+    add_versions("v2.0.1", "62071ac22872d9b936408e4a067d15edcdd61dce864ace8725eacdaefe23b898")
+    add_versions("v2.0", "1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c")
+    add_versions("v1.6.4", "9ae6a5ffa77462fee4fc34c24a7c2459a2a5851f")
+
+    add_deps("autoconf", "automake", "libtool", "pkg-config")
+    add_cxxflags("--std=c++17")
```
Comment (waruqi) [inline packages/l/libgpiod/xmake.lua:12]: please remove it

File: packages/l/libgpiod/xmake.lua [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+    on_install("macos", "linux", function (package)
+        local configs = {}
+        table.insert(configs, "--enable-tools=yes")
+        table.insert(configs, "-enable-bindings-cxx")
+        table.insert(configs, "--prefix="..package:installdir())
```
Comment (waruqi) [inline packages/l/libgpiod/xmake.lua:18]: please remove this line

File: packages/l/libgpiod/xmake.lua [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    add_urls("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-$(version).tar.gz",
+             "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git")
+    add_versions("v2.0.1", "62071ac22872d9b936408e4a067d15edcdd61dce864ace8725eacdaefe23b898")
+    add_versions("v2.0", "1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c")
+    add_versions("v1.6.4", "9ae6a5ffa77462fee4fc34c24a7c2459a2a5851f")
+
+    add_deps("autoconf", "automake", "libtool", "pkg-config")
+    add_cxxflags("--std=c++17")
+
+    on_install("macos", "linux", function (package)
```
Comment (waruqi) [inline packages/l/libgpiod/xmake.lua:14]: "macosx"

File: packages/l/libgpiod/xmake.lua (hunk 8 above)
Comment (waruqi) [reply packages/l/libgpiod/xmake.lua:18]: if it is library, you need switch static/shared configuration.

File: packages/l/libgpiod/xmake.lua [hunk 10]
```diff
@@ -0,0 +1,24 @@
+package("libgpiod")
+    set_homepage("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/about/")
+    set_description("libgpiod - C library and tools for interacting with the linux GPIO character device (gpiod stands for GPIO device)")
+    
+    add_urls("https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-$(version).tar.gz",
+             "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git")
+    add_versions("v2.0.1", "62071ac22872d9b936408e4a067d15edcdd61dce864ace8725eacdaefe23b898")
+    add_versions("v2.0", "1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c")
```
Comment (waruqi) [inline packages/l/libgpiod/xmake.lua:8]: why not use sha256?

Comment (Greeley) [reply packages/l/libgpiod/xmake.lua:8]: v2.0.1 is a sha256 retrieved with `shasum -a 256 libgpiod-2.0.1.tar.gz`

File: packages/l/libgpiod/xmake.lua [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    on_install("macos", "linux", function (package)
+        local configs = {}
+        table.insert(configs, "--enable-tools=yes")
+        table.insert(configs, "-enable-bindings-cxx")
+        table.insert(configs, "--prefix="..package:installdir())
+        import("package.tools.autoconf").install(package, configs)
+    end)
+
+    on_test(function (package)
+        os.runv("gpiodetect")
```
[the remark this answers is not in the document]
Comment (Greeley) [reply packages/l/libgpiod/xmake.lua:23]: docs?

File: packages/l/libgpiod/xmake.lua (hunk 10 above)
Comment (waruqi) [reply packages/l/libgpiod/xmake.lua:8]: but v2.0 (1e4dbbe8b3f32adf8818f1b0dfd322876aa9477c) is not sha256 

```
shasum -a 256 ~/Downloads/libgpiod-v2.0.tar.gz
a0f835c4ca4a2a3ca021090b574235ba58bb9fd612d8a6051fb1350054e04fdd  /Users/ruki/Downloads/libgpiod-v2.0.tar.gz
```

File: packages/l/libgpiod/xmake.lua (hunk 11 above)
[the remark this answers is not in the document]
Comment (waruqi) [reply packages/l/libgpiod/xmake.lua:23]: https://xmake.io/#/manual/package_dependencies?id=packageon_test

File: packages/l/libgpiod/xmake.lua (hunk 10 above)
Comment (Greeley) [reply packages/l/libgpiod/xmake.lua]: Does it matter? the sha256 one doesn't even validate, why would I put effort into the other 2 if the first one doesn't actually verify?

**Pull request merged**